plan/client_test.go

```diff
@@ -1,193 +1,56 @@
 package plan
 
 import (
-	"fmt"
 	"testing"
 
+	assert "github.com/stretchr/testify/require"
 	stripe "github.com/stripe/stripe-go"
 	"github.com/stripe/stripe-go/currency"
-	. "github.com/stripe/stripe-go/utils"
+	_ "github.com/stripe/stripe-go/testing"
 )
 
-func init() {
-	stripe.Key = GetTestKey()
-}
-
-func TestPlanNew(t *testing.T) {
-	planParams := &stripe.PlanParams{
-		ID:            "test_plan",
-		Name:          "Test Plan",
-		Amount:        99,
-		Currency:      currency.USD,
-		Interval:      Month,
-		IntervalCount: 3,
-		TrialPeriod:   30,
-		Statement:     "Test Plan",
-	}
-
-	target, err := New(planParams)
-
-	if err != nil {
-		t.Error(err)
-	}
-
-	if target.ID != planParams.ID {
-		t.Errorf("ID %q does not match expected id %q\n", target.ID, planParams.ID)
-	}
-
-	if target.Name != planParams.Name {
-		t.Errorf("Name %q does not match expected name %q\n", target.Name, planParams.Name)
-	}
-
-	if target.Amount != planParams.Amount {
-		t.Errorf("Amount %v does not match expected amount %v\n", target.Amount, planParams.Amount)
-	}
-
-	if target.Currency != planParams.Currency {
-		t.Errorf("Currency %q does not match expected currency %q\n", target.Currency, planParams.Currency)
-	}
-
-	if target.Interval != planParams.Interval {
-		t.Errorf("Interval %q does not match expected interval %q\n", target.Interval, planParams.Interval)
-	}
-
-	if target.IntervalCount != planParams.IntervalCount {
-		t.Errorf("Interval count %v does not match expected interval count %v\n", target.IntervalCount, planParams.IntervalCount)
-	}
-
-	if target.TrialPeriod != planParams.TrialPeriod {
-		t.Errorf("Trial period %v does not match expected trial period %v\n", target.TrialPeriod, planParams.TrialPeriod)
-	}
-
-	if target.Statement != planParams.Statement {
-		t.Errorf("Statement %q does not match expected statement %q\n", target.Statement, planParams.Statement)
-	}
-
-	Del(planParams.ID)
+func TestPlanDel(t *testing.T) {
```

Comment:
The client tests represent one of the largest sections of change: instead of doing elaborate and brittle setup to make slow calls against the API that don't run in CI because they fail intermittently and are slow, we just do the simplest possible calls against the API stub.

The stub will perform some basic parameter validation, but in general is far more forgiving: it'll give us back a resource no matter what ID we asked for (and generally most of all operations will succeed), so we can significantly shorten these tests.

~200 lines down to ~50 and probably on the order of 3-4 orders of magnitude faster. The only downside is that we don't get to test the API itself, which is fine.